### ISSUE TITLE
How to use 2 PRNGKeys to generate a new PRNGKey?

### ISSUE THREAD

**Issue by cccntu** [google/jax#5549]
I want to `vmap` on 2  dimensions.

I saw an approach like this, but it seems quite verbose and c will only have 2^32 possibilities)
```
a = jax.random.split(rng_1)[0, 0]
b = jax.random.split(rng_2)[0, 0]
c = jax.random.PRNGKey(a+b)
```
What's a correct way to do this? It would be cool if we can have a `jax.random.merge((rng_1, rng_2))`. 

Actually I found `jax.random.split((rng_1, rng_2))` seems to work, but it's not documented and doesn't accepts more than 2 rngs. Can someone explain this to me? Thanks!

**Comment by tomhennigan** [comment]
Random keys are vectors of uint32 so you could do `c = a + b` to get a new key from a and b (there are probably some edge cases where you'll get back a or b, e.g. if either of the keys are `jax.random.PRNGKey(0)`).

It might be useful if you could post a simplified version of how you are planning to use these keys, I'm wondering if there is any benefit to merging the keys or if instead you could just discard one. Alternatively can you pass in a single block of keys to your double vmap'd function:

```python
def f(key):
  return jax.random.normal(key, [])

i = 2
j = 2
key = jax.random.PRNGKey(42)
key = jax.random.split(key, i * j).reshape((i, j) + key.shape)
jax.vmap(jax.vmap(f))(key)
# DeviceArray([[-0.68319297,  1.4919369 ],
#              [ 2.5348368 , -0.8791437 ]], dtype=float32)
```

**Comment by cccntu** [comment]
Thanks for the timely reply! 
I think your solution works and there is indeed no need to merge keys.

For completeness, here is what I wanted to do:
I was inspired by this blog post that trains multiple models in parallel, each using different bootstrapped dataset. http://willwhitney.com/parallel-training-jax.html

It uses a key and `randint` to map from a large space to the data index space, for bootstrap sampling without keeping the samples. 

I wanted to modify it so it's not bootstrapped, instead sampled uniformly with replacement. I guess I can just use your approach.

**Comment by mattjj** [comment]
While you can just add PRNG keys, I wouldn't recommend that. It's a bit of a data abstraction violation, and could lead to more key collisions than if you just handle things with `split`.

I like the idea of generating a single block of keys, and also I like the reshaping strategy. Another way to do it is just to split twice, like `split(split(key, j), i)`. Each split will add a leading axis of the appropriate size. The reshape approach is likely faster though.

I think we covered this question, so I'll close this issue.
